### PR TITLE
Fix tpt

### PR DESCRIPTION
**Pull request by cxhernandez**
- [x] Implement feature / fix bug
 - [x] Add tests
 - [x] Update changelog

add logic to handle MSMs with `msm.all_populations_`

### CODE REVIEW

**Comment by msultan** [comment]
i am not sure this is the right way to do this. the mean of all the transition matrices in not necessarily a valid transition matrix. imo, it would be better to iterate over all the matrices and then return the mean of the transition paths.

**Comment by cxhernandez** [comment]
What's the mean of a transition path....

**Comment by msultan** [comment]
yea fair point. btw does this work if you have ergodic trimming in some of the matrices?

**Comment by cxhernandez** [comment]
I don't really know much about this code, but I don't know why it wouldn't. It might not report the correct state mappings though....

**Comment by msultan** [comment]
yea if there is trimming this is gonna give completely weird results.

**Comment by cxhernandez** [comment]
Alright, I can add a fix for that— shouldn't be too bad.

**Comment by msultan** [comment]
hmm, how about returning lists of tpts +fluxes for each of the matrices and letting the modeller decide what to with them?  
also, if a state is thrown out in model but not in another i am not even sure what tpt should even do in that context.

**Comment by cxhernandez** [comment]
> if a state is thrown out in one model 

That's only an issue in bootstrapped msms, right? I'm not going to worry about that now....

**Comment by msultan** [comment]
>That's only an issue in bootstrapped msms, right?

yea, but the all_tmats_ attribute doesnt map the transition matrix back to the mle msm labels. its just all the data.

File: msmbuilder/tpt/committor.py [hunk 1]
```diff
@@ -77,7 +77,10 @@ def committors(sources, sinks, msm):
     sinks = np.array(sinks, dtype=int).reshape((-1, 1))
 
     n_states = msm.n_states_
-    tprob = msm.transmat_
+    if hasattr(msm, 'all_transmats_'):
+        tprob = msm.all_transmats_.mean(0)
```
Comment (mpharrigan) [inline msmbuilder/tpt/committor.py]: is this `axis=0`? Can you add the keyword so readers of the code won't be confused

**Comment by msultan** [comment]
oh this is for bayesian msms only? i used the same name for the bootstrap version and was thinking of that. the bayesian ones shouldnt have the mapping problem. 

i think adding an explicit isinstance(msm, bayesianmsm) is necessary. i am still not sure if the mean of the tmat is the right statistic to calculate.

**Comment by cxhernandez** [comment]
Yeah I had bayesian msms in mind. IMO we should fix bootstrap msm so it behaves more like bayesian msm. It's not good to keep having to add checks like this...

**Comment by msultan** [comment]
i thought about that but without having to assume underlying structure on the model, its not easy. its definitely doable not by adding some prior counts to each msm.

**Comment by cxhernandez** [comment]
How about something like this:

```python

def func(source, sink, msm_copy):

    if hasattr(msm, 'all_transmats_'):
        output = np.zeros_like(msm.all_transmats_)
        for i, tprob in enumerate(msm.all_transmats_):
            msm_copy = copy.copy(msm)
            msm_copy.transmat_ = tprob
            del(msm_copy.__dict__["all_transmats_"])
            output[i, :, :] = func(source, sink, msm_copy)
        return output
    else:
        tprob = msm.transmat_

    # execute code normally

```

So it returns an array of whatever it's measuring (e.g. committor values, fluxes) with an entry for each `transmat_`. You or @mpharrigan might find this hacky though....

**Comment by cxhernandez** [comment]
Or we could have private functions that operate on the `transmat_` directly and just have the public functions figure out the logic...

**Comment by cxhernandez** [comment]
@msmbuilder/developers: This should be ready for review whenever you get a chance

File: devtools/conda-recipe/meta.yaml [hunk 2]
```diff
@@ -47,7 +47,7 @@ test:
     - msmbuilder
   commands:
     - msmb -h
-    - nosetests msmbuilder -v --with-timer
+    - nosetests msmbuilder -v --with-timer --timer-top-n 5
```
Comment (mpharrigan) [inline devtools/conda-recipe/meta.yaml]: why is this showing up?

File: msmbuilder/decomposition/ktica.py [hunk 3]
```diff
@@ -105,11 +105,7 @@ class KernelTICA(tICA):
     def __init__(self, n_components=None, lag_time=1, shrinkage=None,
                  kinetic_mapping=False, kernel='rbf', degree=3, gamma=None,
                  coef0=1., stride=1, landmarks=None, random_state=None,
-<<<<<<< HEAD
                  kernel_params=None):
-=======
```
Comment (mpharrigan) [inline msmbuilder/decomposition/ktica.py]: why is this showing up?

**Comment by mpharrigan** [comment]
Can you merge master? There are weird things showing up in the diff

**Comment by msultan** [comment]
@cxhernandez as far as i can tell you are now gonna report the mean of the mftps, hub scores, commitors values for each state/path across all the transition matrices. Here are my two main questions. 

1. If the top path changes across the transition models, is it fair to compute the mean of the mfpts then? 

2. Based upon the convos we have had about robust statistics, is mean the right statistic to report? Perhaps the median value would take care of outliers.

**Comment by cxhernandez** [comment]
> If the top path changes across the transition models, is it fair to compute the mean of the mfpts then?

It's still a valid mean IMO. The whole point is that you're trying to get a gauge of the average over many similar models.

> Based upon the convos we have had about robust statistics, is mean the right statistic to report? Perhaps the median value would take care of outliers.

This is a good point. It might be better to report the median.

**Pull request merged**